### PR TITLE
Link Tokenizer Info to Model

### PR DESCRIPTION
**Pull request by jonthegeek**
Major changes to make models safer.

### CODE REVIEW

**Review by jonathanbratt — CHANGES_REQUESTED** [review]
looking good! Just a few things that we mostly already discussed.

File: R/luz_callbacks.R [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  luz_callback_bert_tokenize_generator <- luz::luz_callback(
+    "bert_tokenize_callback",
+    initialize = function(submodel_name = NULL,
+                          n_tokens = NULL,
+                          verbose = TRUE) {
+      self$submodel_name <- submodel_name
+      self$n_tokens <- n_tokens
+      self$verbose <- verbose
+    },
+    on_fit_begin = function() {
```
[the remark this answers is not in the document]
Comment (jonthegeek) [reply R/luz_callbacks.R:53]: There are other initialization things that run on_fit_begin and I think it's fine for this to run at the default level.

File: R/luz_callbacks.R [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      # The ctx object contains everything being used by luz. The specific
+      # things we want are model, train_data, and valid_data. Make sure that
+      # those datasets are tokenized to match this model.
+      model <- ctx$model
+      if (!is.null(self$submodel_name)) {
+        model <- model[[self$submodel_name]]
```
[the remark this answers is not in the document]
Comment (jonthegeek) [reply R/luz_callbacks.R:59]: By default this should be NULL. It's only not NULL if you wrap the pretrained model inside another model, so we can't set a default (other than NULL).

[the remark this answers is not in the document]
Comment (jonathanbratt) [reply R/luz_callbacks.R:78]: Ah, that makes sense. I still think maybe in the future we could make this argument less necessary, but this is fine for now.

**Review by jonathanbratt — APPROVED** [review]

**Pull request merged**